examples/linear_memory/src/lib.rs

```diff
@@ -10,20 +12,20 @@ impl Contract {
         e.binary_new_from_linear_memory(buf.as_ptr() as u32, len)
     }
 
-    pub fn from_guest(e: Env, b: Binary, b_pos: u32, lm_pos: u32, len: u32) -> Binary {
+    pub fn from_guest(e: Env, b: Binary, b_pos: u32, buf_off: u32, len: u32) -> Binary {
         let buf: [u8; 4] = [0, 1, 2, 3];
-        assert!(lm_pos + len > buf.len() as u32);
```

Comment:
Welp this is embarrassing. 
What I don't understand is how did this pass the [test](https://github.com/stellar/rs-stellar-contract-env/blob/main/stellar-contract-env-host/src/test/binary.rs#L168-L182)? They clearly did not panic or trap.